### PR TITLE
Pimp module

### PR DESCRIPTION
**Pull request by ngzhian**
Partially address #317, under module details section (i don't really want to add the like and share buttons cos i'm not a fan, so that can come later). The rest of the implementation will be in separate PRs.

Added the exam dates, semesters offered, official ivle and cors links, and add to sem/remove to sem buttons.

When module is not in timetable you can add it to the semesters offered:
screen shot 2016-12-08 at 1 12 47 am

If it is already in the timetable you can remove:
screen shot 2016-12-08 at 1 14 56 am

### CODE REVIEW

File: v3/src/js/utils/date.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// @flow
+export function dateForDisplay(date: String): string {
```
Comment (yangshun) [inline v3/src/js/utils/date.js]: We already have a date display function called `formatExamDate` in `modules.js` that we use to display on the timetable page. Should use that for consistency sake.

File: v3/src/js/views/browse/ModulePageContainer.jsx [hunk 2]
```diff
@@ -34,12 +42,68 @@ export class ModulePageContainer extends Component {
...
+        .sort((a, b) => a.Semester - b.Semester)
+        .map(h => ({ semester: h.Semester, date: h.ExamDate })))
+      : [];
+  }
+
+  moduleHasBeenAdded(module: Module, semester: num): boolean {
+    const timetables = this.props.timetables;
+    return (
+      timetables[semester] &&
+      typeof timetables[semester][module.ModuleCode] !== 'undefined'
```
Comment (yangshun) [inline v3/src/js/views/browse/ModulePageContainer.jsx]: Would `!!timetables[semester][module.ModuleCode]` work?

File: v3/src/js/views/browse/ModulePageContainer.jsx [hunk 3]
```diff
@@ -34,12 +42,68 @@ export class ModulePageContainer extends Component {
...
+
+    const semsOffered = this.semestersOffered()
+      .map(sem => `Semester ${sem}`)
+      .join(', ');
+
+    const addOrRemoveToTimetableLinks = this.semestersOffered().map(
+      semester => (
+        this.moduleHasBeenAdded(module, semester) ?
+          (
+            <RemoveModuleButton key={semester} semester={semester} onClick={() =>
```
Comment (yangshun) [inline v3/src/js/views/browse/ModulePageContainer.jsx]: The parens around the component are redundant

File: v3/src/js/utils/date.js (hunk 1 above)
Comment (ngzhian) [inline v3/src/js/utils/date.js]: I was looking for something like that and couldn't find :( Thanks for pointing this out!

File: v3/src/js/views/browse/ModulePageContainer.jsx (hunk 2 above)
Comment (ngzhian) [inline v3/src/js/views/browse/ModulePageContainer.jsx]: yes it would, thanks!

**Pull request merged**